src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -1073,6 +1073,15 @@ _state_machine_run (_state_machine_t *state_machine, bson_t *result, bson_error_
          break;
       case MONGOCRYPT_CTX_DONE:
          goto success;
+         break;
+      case MONGOCRYPT_CTX_NEED_MONGO_COLLINFO_WITH_DB:
+         bson_set_error (error,
+                         MONGOC_ERROR_CLIENT_SIDE_ENCRYPTION,
+                         0,
```

Comment:
```suggestion
                         MONGOC_ERROR_CLIENT_INVALID_ENCRYPTION_STATE,
```

Suggest using a non-zero error code. Callers may assume if the code is 0, no error occurred.